sherpa/utils/src/_psf.cc

```diff
@@ -250,6 +259,18 @@ static int _convolve( tcdPyData* self, double* source, double* kernel,
   return EXIT_SUCCESS;
 }
 
+static bool same_size_arrays( npy_intp size1, npy_intp size2,
+                              const char *suffix, const char *name1,
+                              const char *name2 ) {
+  if ( size1 != size2 ) {
+    std::ostringstream err;
+    err << "input array sizes do not match" << suffix;
+    err << name1 << size1 << name2 << size2;
```

Comment:
The error should contain all of the formatting- all of the calls to same_size_array are passing in the variable names and formatting text via the suffix, nam1, name2 parameters.  For example:

     err << name1 << ": " << size1 << " vs " << name2 << ": " << size2;

and the arguments passed in as name1 and 2 in the references (line 298, 302, etc...) should only pass in the actual names. Doing so, centralizes the formatting to one location so that it is easier to change and also makes the variable names accurately reflect their purpose.

Reply:
done